### PR TITLE
Update checkout action version, add CODEOWNERS

### LINKED ISSUE #12: Update GitHub actions
- Rename `master` branch to `main`.
- Update GitHub actions to use `actions/checkout@v3` and to run on PRs as well as changes to `main`.
- Add CODEOWNERS file

Once that is complete, update branch protection for `main` to require all unit tests to pass.

### PR DESCRIPTION
**Pull request by JacobHearst**
Closes #12 . Will update `master` to `main` after this is approved and merged

### CODE REVIEW

**Comment by JacobHearst** [comment]
Is the rename of `master` to `main` something that needs to be communicated somehow? Anyone with a local copy of this repo will need to make some changes/re-clone. Just not sure how large of an impact that really is

File: CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1 @@
+* @jamf/apple-natives-write
```
Comment (macblazer) [inline CODEOWNERS:1]: Generally these kind of text files have a newline at the end of them.

File: .github/workflows/build-and-test.yml
Comment (macblazer) [inline .github/workflows/build-and-test.yml:1]: Issue #12 also asked for this workflow to be updated to run on pull requests to main as well as pushes to main.  Is this something that will have a new PR after this is merged, or do you want to tackle it here?

Comment (JacobHearst) [reply .github/workflows/build-and-test.yml:1]: Ah, missed the push to main bit. Will fix

**Review by macblazer — APPROVED** [review]

**Pull request merged**